surveys/promise-rejections/survey.md

```diff
@@ -0,0 +1,152 @@
+
+## What is a Promise rejection?
+
+A Promise rejection indicates that something went wrong while executing a Promise or an async function. Rejections can occur on several situations: thorwing inside an async function or a `Promise` executor/then/catch/finally callback, when calling the `reject` callback of an `executor`, or when calling `Promise.reject`.
+
+```js
+
+Promise.reject()  // This will result in a rejection
+
+new Promise((fulfill, reject) => {
+  reject();  // This will result in a rejection
+});
+
+new Promise(() => {
+  throw new Error();  // This will result in a rejection
+}).then(() => {
+  throw new Error();  // This will result in a rejection
+}, () => {
+  throw new Error();  // This will result in a rejection
+}).catch(() => {
+  throw new Error();  // This will result in a rejection
+}).finally(() => {
+  throw new Error();  // This will result in a rejection
+});
+
+async function () {
+  throw new Error();  // This will result in a rejection
+}
+
+async function () {
+  if (a === undefined)
+    a();  // This will result in a rejection
+}
+```
+
+## What is an unhandled rejection?
+
+Rejections are designed to be handled asynchronously. In JavaScript, Promises can be handled when awaiting for a promise inside a try/catch block, or via .catch calls:
+
+
+```js
+async function foo() {
+  throw new Error();
+}
+
+foo().catch(() => console.error("an error occured"));  // 1. Handled
+
+try {
+  await foo();  // 2. Handled
+} catch(e) {
+  console.error("an error occured")
+}
+
+foo(); // 3. Unhandled, but execution continues
+
+const rejected = foo(); // 4. Unhandled on current event loop tick
+
+// 4. ... but handled asynchronously when the setTimeout callback is popped from
+// the event loop
+setTimeout(() => rejected.catch(() => console.error("an error occured")), 100);
+```
+
+An unhandled rejection is a rejection which hasn't been handled yet. It might be handled in the future, like example 4, but it might also stay unhandled forever (like example 3).
+
+## Are you currently using Promises, async/await, a mix, or neither?
+
+Assume using as creating your own `new Promise`, `Promise.resolve`, and `async function`; or using libraries that produce Promises (that you use with `.then`/`.catch`, or with `.await`)
+
+(Multiple choice)
+
+  - [ ] Promises
+  - [ ] async/await
+  - [ ] Not writing Promise-based code or using Promise-based libraries
+
+## How are you handling rejections today?
+
+When consuming Promises, async functions or thenables, which of the options below do you use to handle rejections?
+
+(Multiple choice)
+
+  - [ ] `.catch()`
+  - [ ] `try / catch` wrapping an `await` operation
+  - [ ] Ignore the rejection
+  - [ ] Not writing Promise-based code or using Promise-based libraries
+
+## Do you know that Node.js has a global handled for unhandled rejections (`process.on('unhandledRejection')`)? If so, do you use it?
```

Comment:
Can we please add (for comparison) the same question about `uncaughtException`? I want to know if the people answering aren't aware of the promises hook are also not aware of the non-promises hook.